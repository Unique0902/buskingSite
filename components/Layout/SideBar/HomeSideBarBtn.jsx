import Link from 'next/link';
import React from 'react';

const HomeSideBarBtn = ({ url, children }) => {
  return (
    <li className='flex flex-row hover:opacity-70 hover:bg-gray-300'>
      <Link
        href={url}
        className='flex-1 px-6 py-6 font-sans text-2xl font-semibold'
      >
        {children}
      </Link>
    </li>
  );
};

export default HomeSideBarBtn;
