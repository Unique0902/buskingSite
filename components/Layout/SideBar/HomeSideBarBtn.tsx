import Link from 'next/link';
import React, { ReactNode } from 'react';
type Props = {
  url: string;
  children: ReactNode;
};
const HomeSideBarBtn = ({ url, children }: Props) => {
  return (
    <li className='flex flex-row hover:opacity-70 hover:bg-gray-300'>
      <Link
        href={url}
        className='flex-1 px-6 py-6 font-sans text-2xl font-semibold'
      >
        {children}
      </Link>
    </li>
  );
};

export default HomeSideBarBtn;
