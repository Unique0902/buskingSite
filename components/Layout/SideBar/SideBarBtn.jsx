import Link from 'next/link';
import React from 'react';

export default function SideBarBtn({
  name,
  selectedBtn,
  isHide,
  text,
  children,
}) {
  const btnStyle =
    'text-white flex items-center pl-5 py-4 font-sans w-full text-left cursor-pointer font-medium text-lg hover:bg-zinc-600 ';
  const hideBtnStyle =
    'text-white flex items-center py-4 font-sans w-full cursor-pointer font-medium text-lg hover:bg-zinc-600 justify-center';
  return (
    <Link
      href={`/app/${name}`}
      className={`${isHide ? hideBtnStyle : btnStyle} ${
        selectedBtn === name ? 'text-blue-400' : 'text-white'
      } `}
    >
      {React.cloneElement(children, {
        color: `${selectedBtn === name ? '#60a5fa' : 'white'}`,
      })}
      {!isHide && text}
    </Link>
  );
}
