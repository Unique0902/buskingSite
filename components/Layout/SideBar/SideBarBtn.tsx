import React from 'react';

import Link from 'next/link';

import { iconName } from '../../../assets/icon/constants';
import Icon from '../../../assets/icon/icon';
type Props = {
  name: string;
  selectedBtn: string;
  isHide: boolean;
  text: string;
  icon: iconName;
};
export default function SideBarBtn({
  name,
  selectedBtn,
  isHide,
  text,
  icon,
}: Props) {
  const btnStyle =
    'text-white flex items-center pl-5 py-4 w-full text-left cursor-pointer font-medium text-lg hover:bg-zinc-600 ';
  const hideBtnStyle =
    'text-white flex items-center py-4 w-full cursor-pointer font-medium text-lg hover:bg-zinc-600 justify-center';

  return (
    <Link
      href={`/app/${name}`}
      className={`${isHide ? hideBtnStyle : btnStyle} ${
        selectedBtn === name ? 'text-blue-400' : 'text-white'
      } `}
    >
      <div className={`${!isHide && 'mr-4'}`}>
        <Icon
          size={24}
          color={`${selectedBtn === name ? '#60a5fa' : '#FFFFFF'}`}
          icon={icon}
        />
      </div>

      {!isHide && text}
    </Link>
  );
}
