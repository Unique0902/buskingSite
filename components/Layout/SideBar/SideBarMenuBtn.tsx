import React from 'react';

import Link from 'next/link';

import { iconName } from '../../../assets/icon/constants';
import Icon from '../../../assets/icon/icon';
type Props = {
  name: string;
  isSelected: boolean;
  isMiniMode: boolean;
  text: string;
  icon: iconName;
};
export default function SideBarMenuBtn({
  name,
  isSelected,
  isMiniMode,
  text,
  icon,
}: Props) {
  const btnStyle =
    'text-white flex items-center pl-5 py-4 w-full text-left cursor-pointer font-medium text-lg hover:bg-zinc-600 ';
  const hideBtnStyle =
    'text-white flex items-center py-4 w-full cursor-pointer font-medium text-lg hover:bg-zinc-600 justify-center';

  return (
    <Link
      href={`/app/${name}`}
      className={`${isMiniMode ? hideBtnStyle : btnStyle} ${
        isSelected ? 'text-blue-400' : 'text-white'
      } `}
      data-testid={isSelected ? 'selectedMenu' : ''}
    >
      <div className={`${!isMiniMode && 'mr-4'}`}>
        <Icon
          size={24}
          color={`${isSelected ? '#60a5fa' : '#FFFFFF'}`}
          icon={icon}
        />
      </div>

      {!isMiniMode && text}
    </Link>
  );
}
