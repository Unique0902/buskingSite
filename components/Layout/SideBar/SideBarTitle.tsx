import React from 'react';
import Icon from '../../../assets/icon/icon';

interface Props {
  isMiniMode: boolean;
}

const SideBarTitle: React.FC<Props> = ({ isMiniMode }: Props) => {
  if (isMiniMode) {
    return (
      <div className='flex items-center justify-center w-full py-4 text-2xl text-center text-blue-600 border-b border-gray-600 border-solid'>
        <Icon size={32} color='#60a5fa' icon='Book' />
      </div>
    );
  }
  return (
    <div className='flex items-center gap-3 px-5 py-3 text-white border-b border-gray-600 border-solid'>
      <Icon size={32} color='#60a5fa' icon='Book' />
      <p className='text-2xl font-semibold '>노래책</p>
    </div>
  );
};

export default SideBarTitle;
