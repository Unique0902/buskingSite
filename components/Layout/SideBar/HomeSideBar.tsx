import React, { useEffect, useRef } from 'react';
import { ArrowRightIcn, BookIcn } from '../../../assets/icon/icon';
import HomeSideBarBtn from './HomeSideBarBtn';
type Props = {
  setIsShowSideBar: React.Dispatch<React.SetStateAction<boolean>>;
};
const HomeSideBar = ({ setIsShowSideBar }: Props) => {
  const wrapperRef = useRef<HTMLDivElement>();

  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      if (
        wrapperRef.current &&
        !wrapperRef.current.contains(event.target as Node)
      ) {
        setIsShowSideBar(false);
      }
    };
    document.addEventListener('mousedown', handleClickOutside);
    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, [wrapperRef, setIsShowSideBar]);

  return (
    <>
      <aside
        ref={wrapperRef}
        className={` bg-gray-100 text-black relative max-lg:absolute right-0 top-0 max-lg:h-full max-lg:z-40 w-64`}
      >
        <div className='flex items-center px-6 py-6 text-black border-b border-gray-600 border-solid'>
          <BookIcn
            width={32}
            height={32}
            color={'#60a5fa'}
            className={`mr-4`}
          />
          <h1 className='font-sans text-2xl font-semibold '>노래책</h1>
        </div>

        <ul className='flex flex-col'>
          <HomeSideBarBtn url={'/'}>소개</HomeSideBarBtn>
          <HomeSideBarBtn url={'/'}>찾기</HomeSideBarBtn>
          <HomeSideBarBtn url={'/'}>지원</HomeSideBarBtn>
          <HomeSideBarBtn url={'/'}>다운로드</HomeSideBarBtn>

          <li>
            <button
              className='absolute bottom-0 flex justify-end w-full pr-5 text-right text-black border-t border-gray-600 border-solid hover:opacity-70 py-7'
              onClick={() => {
                setIsShowSideBar(false);
              }}
            >
              <ArrowRightIcn width={20} height={20} color={'black'} />
            </button>
          </li>
        </ul>
      </aside>
    </>
  );
};

export default HomeSideBar;
