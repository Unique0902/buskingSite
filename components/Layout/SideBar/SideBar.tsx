import React, { useState, useRef } from 'react';

import { useRouter } from 'next/router';

import SideBarMenuBtn from './SideBarMenuBtn';
import { SideBarMenuSectionData } from '../../../store/data/SideBarMenus';
import SideBarTitle from './SideBarTitle';
import SideBarToggleBtn from './SideBarToggleBtn';
import { useClickOutside } from '../../../hooks/useClickOutside';
import { useSideBarMenu } from '../../../hooks/useSideBarMenu';

interface Props {
  setIsShowSideBar: React.Dispatch<React.SetStateAction<boolean>>;
  sideBarMenuSectionDataArr: SideBarMenuSectionData[];
  isSmScreen: boolean;
}

const SideBar = ({
  setIsShowSideBar,
  sideBarMenuSectionDataArr,
  isSmScreen,
}: Props) => {
  const [isHide, setIsHide] = useState<boolean>(false);
  const wrapperRef = useRef<HTMLDivElement>(null);
  const router = useRouter();
  const { sideBarMenuSectionArr } = useSideBarMenu(
    sideBarMenuSectionDataArr,
    router.pathname
  );

  useClickOutside(wrapperRef, () => setIsShowSideBar(false), isSmScreen);

  return (
    <>
      <aside
        ref={wrapperRef}
        className={` bg-zinc-800 dark:border-r dark:border-r-gray-600 relative max-lg:absolute max-lg:h-full max-lg:z-40 ${
          isHide ? 'w-16' : 'w-64'
        }`}
      >
        <SideBarTitle isMiniMode={isHide} />

        <ul className='flex flex-col'>
          {sideBarMenuSectionArr.map((secData, secIdx) => (
            <React.Fragment key={'sideBarMenuSec' + secIdx}>
              {secData.title && !isHide && (
                <li>
                  <p className='pt-3 pb-3 pl-5 text-sm text-gray-400 border-t border-gray-600 border-solid'>
                    {secData.title}
                  </p>
                </li>
              )}
              {secData.data.map((menuData) => (
                <SideBarMenuBtn
                  key={menuData.nameArr[0]}
                  name={menuData.nameArr[0]}
                  isSelected={menuData.isSelected}
                  isHide={isHide}
                  text={menuData.text}
                  icon={menuData.icon}
                />
              ))}
            </React.Fragment>
          ))}
        </ul>
        <SideBarToggleBtn
          isHide={isHide}
          isSmScreen={isSmScreen}
          setIsHide={setIsHide}
          setIsShowSideBar={setIsShowSideBar}
        />
      </aside>
    </>
  );
};

export default SideBar;
