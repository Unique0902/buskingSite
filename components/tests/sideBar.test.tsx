import { screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { expect, describe, it, jest } from '@jest/globals';
import SideBar from '../Layout/SideBar/SideBar';
import { testSideBarMenuSectionDataArr } from '../../tests/testData';

describe('SideBar component test', () => {
  const setIsShowSideBar = jest.fn();
  it('renders correctly', () => {
    const component = renderer.create(
      <SideBar
        setIsShowSideBar={setIsShowSideBar}
        sideBarMenuSectionDataArr={testSideBarMenuSectionDataArr}
        isSmScreen={false}
        routerPathName={'/app/testMenu1'}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('check sidebar to be miniMode when click toggle btn', async () => {
    render(
      <SideBar
        setIsShowSideBar={setIsShowSideBar}
        sideBarMenuSectionDataArr={testSideBarMenuSectionDataArr}
        isSmScreen={false}
        routerPathName={'/app/testMenu1'}
      />
    );
    expect(screen.queryByText('노래책')).not.toBeNull();

    const toggleBtn = screen.getByTestId('toggleBtn');
    await userEvent.click(toggleBtn);
    expect(screen.queryByText('노래책')).toBeNull();
  });

  //   it('check sidebar to hide when screen is small', async () => {
  //     render(
  //       <SideBar
  //         setIsShowSideBar={setIsShowSideBar}
  //         sideBarMenuSectionDataArr={testSideBarMenuSectionDataArr}
  //         isSmScreen={true}
  //         routerPathName={'/app/testMenu1'}
  //       />
  //     );

  //     const outerBox = screen.getByText('outer');
  //     await userEvent.click(outerBox);
  //     expect(handleClickOther).toHaveBeenCalled();
  //   });

  it('check sidebar to hide when screen is small and click outside', async () => {
    render(
      <>
        <SideBar
          setIsShowSideBar={setIsShowSideBar}
          sideBarMenuSectionDataArr={testSideBarMenuSectionDataArr}
          isSmScreen={true}
          routerPathName={'/app/testMenu1'}
        />
        <div>outer</div>
      </>
    );

    const outerBox = screen.getByText('outer');
    await userEvent.click(outerBox);
    expect(setIsShowSideBar).toHaveBeenCalledWith(false);
  });

  it('check sidebarMenuBtn testMenu1 to activate when testMenu1 is selected', async () => {
    render(
      <SideBar
        setIsShowSideBar={setIsShowSideBar}
        sideBarMenuSectionDataArr={testSideBarMenuSectionDataArr}
        isSmScreen={false}
        routerPathName={'/app/testMenu1'}
      />
    );

    const selectedMenu = screen.getByTestId('selectedMenu');
    const testMenu1 = screen.getByText('testMenu1');

    expect(selectedMenu).toBe(testMenu1);
  });

  //   it('check sidebarMenuBtn testMenu2 to activate and testMenu1 to deactivate when click testmenu2', async () => {
  //     render(
  //       <SideBar
  //         setIsShowSideBar={setIsShowSideBar}
  //         sideBarMenuSectionDataArr={testSideBarMenuSectionDataArr}
  //         isSmScreen={false}
  //         routerPathName={'/app/testMenu1'}
  //       />
  //     );

  //     expect(screen.getByTestId('selectedMenu').innerText).toBe('testMenu1');
  //     const testMenu2 = screen.getByText('testMenu2');
  //     await userEvent.click(testMenu2);
  //     expect(screen.getByTestId('selectedMenu').innerText).toBe('testMenu2');
  //   });

  it('check sidebarMenuBtn with various name (testMenu3,testMenu4) to activate when routerPathName is testmenu4', async () => {
    render(
      <SideBar
        setIsShowSideBar={setIsShowSideBar}
        sideBarMenuSectionDataArr={testSideBarMenuSectionDataArr}
        isSmScreen={false}
        routerPathName={'/app/testMenu4'}
      />
    );

    const selectedMenu = screen.getByTestId('selectedMenu');
    const testMenu = screen.getByText('testMenu3,testMenu4');
    expect(selectedMenu).toBe(testMenu);
  });
});
