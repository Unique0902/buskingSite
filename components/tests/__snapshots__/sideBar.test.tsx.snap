// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar component test renders correctly 1`] = `
<aside
  className=" bg-zinc-800 dark:border-r dark:border-r-gray-600 relative max-lg:absolute max-lg:h-full max-lg:z-40 w-64"
>
  <div
    className="flex items-center gap-3 px-5 py-3 text-white border-b border-gray-600 border-solid"
  >
    <svg
      fill="#60a5fa"
      height={32}
      viewBox="0 0 14 14"
      width={32}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.555 11.542c-.446-.523-1.19-1.083-1.91-1.437a6.185 6.185 0 0 0-2.204-.633 6.42 6.42 0 0 1-.285-.031c-.002 0 .002-1.615.01-3.585l.014-3.58.142.013a6.358 6.358 0 0 1 2.597.88 6.43 6.43 0 0 1 1.718 1.602l.106.148v3.402c0 1.87-.008 3.4-.018 3.4a1.27 1.27 0 0 1-.17-.179zm.726-3.23V4.897l.204-.273c.261-.349.786-.873 1.133-1.133a6.028 6.028 0 0 1 3.087-1.204l.147-.015v7.172l-.17.016c-1.608.15-3.002.813-4.092 1.947l-.309.321V8.312zm.996 3.184a5.739 5.739 0 0 1 2.306-1.299c.439-.123 1.01-.21 1.389-.21.199 0 .309-.042.371-.144.047-.077.047-.11.048-3.016V3.889l.087.016c.049.01.186.048.305.086L13 4.06v3.667c0 2.016-.008 3.666-.018 3.666-.01 0-.155-.03-.322-.069a7.224 7.224 0 0 0-1.922-.18 7.3 7.3 0 0 0-2.343.422l-.305.105.187-.176zm-2.672.057a8.439 8.439 0 0 0-1.253-.32 7.06 7.06 0 0 0-1.472-.09 6.633 6.633 0 0 0-1.493.17c-.142.033-.287.066-.323.073L1 11.4V4.046l.276-.08.304-.09c.023-.01.03.59.03 2.935 0 3.206-.008 3.027.13 3.117.047.03.146.045.428.062a5.694 5.694 0 0 1 3.057 1.082c.188.138.638.518.663.56.018.029.006.026-.282-.078z"
        fillRule="nonzero"
      />
    </svg>
    <p
      className="text-2xl font-semibold "
    >
      노래책
    </p>
  </div>
  <ul
    className="flex flex-col"
  >
    <a
      className="text-white flex items-center pl-5 py-4 w-full text-left cursor-pointer font-medium text-lg hover:bg-zinc-600  text-blue-400 "
      data-testid="selectedMenu"
      href="/app/testMenu1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="mr-4"
      >
        <svg
          fill="#60a5fa"
          height={24}
          viewBox="0 0 460.298 460.297"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M230.149 120.939 65.986 256.274c0 .191-.048.472-.144.855-.094.38-.144.656-.144.852v137.041c0 4.948 1.809 9.236 5.426 12.847 3.616 3.613 7.898 5.431 12.847 5.431h109.63V303.664h73.097v109.64h109.629c4.948 0 9.236-1.814 12.847-5.435 3.617-3.607 5.432-7.898 5.432-12.847V257.981c0-.76-.104-1.334-.288-1.707L230.149 120.939z"
            fillRule="nonzero"
          />
          <path
            d="M457.122 225.438 394.6 173.476V56.989c0-2.663-.856-4.853-2.574-6.567-1.704-1.712-3.894-2.568-6.563-2.568h-54.816c-2.666 0-4.855.856-6.57 2.568-1.711 1.714-2.566 3.905-2.566 6.567v55.673l-69.662-58.245c-6.084-4.949-13.318-7.423-21.694-7.423-8.375 0-15.608 2.474-21.698 7.423L3.172 225.438c-1.903 1.52-2.946 3.566-3.14 6.136-.193 2.568.472 4.811 1.997 6.713l17.701 21.128c1.525 1.712 3.521 2.759 5.996 3.142 2.285.192 4.57-.476 6.855-1.998L230.149 95.817l197.57 164.741c1.526 1.328 3.521 1.991 5.996 1.991h.858c2.471-.376 4.463-1.43 5.996-3.138l17.703-21.125c1.522-1.906 2.189-4.145 1.991-6.716-.195-2.563-1.242-4.609-3.141-6.132z"
            fillRule="nonzero"
          />
        </svg>
      </div>
      testMenu1
    </a>
    <a
      className="text-white flex items-center pl-5 py-4 w-full text-left cursor-pointer font-medium text-lg hover:bg-zinc-600  text-white "
      data-testid=""
      href="/app/testMenu2"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="mr-4"
      >
        <svg
          fill="#FFFFFF"
          height={24}
          viewBox="0 0 460.298 460.297"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M230.149 120.939 65.986 256.274c0 .191-.048.472-.144.855-.094.38-.144.656-.144.852v137.041c0 4.948 1.809 9.236 5.426 12.847 3.616 3.613 7.898 5.431 12.847 5.431h109.63V303.664h73.097v109.64h109.629c4.948 0 9.236-1.814 12.847-5.435 3.617-3.607 5.432-7.898 5.432-12.847V257.981c0-.76-.104-1.334-.288-1.707L230.149 120.939z"
            fillRule="nonzero"
          />
          <path
            d="M457.122 225.438 394.6 173.476V56.989c0-2.663-.856-4.853-2.574-6.567-1.704-1.712-3.894-2.568-6.563-2.568h-54.816c-2.666 0-4.855.856-6.57 2.568-1.711 1.714-2.566 3.905-2.566 6.567v55.673l-69.662-58.245c-6.084-4.949-13.318-7.423-21.694-7.423-8.375 0-15.608 2.474-21.698 7.423L3.172 225.438c-1.903 1.52-2.946 3.566-3.14 6.136-.193 2.568.472 4.811 1.997 6.713l17.701 21.128c1.525 1.712 3.521 2.759 5.996 3.142 2.285.192 4.57-.476 6.855-1.998L230.149 95.817l197.57 164.741c1.526 1.328 3.521 1.991 5.996 1.991h.858c2.471-.376 4.463-1.43 5.996-3.138l17.703-21.125c1.522-1.906 2.189-4.145 1.991-6.716-.195-2.563-1.242-4.609-3.141-6.132z"
            fillRule="nonzero"
          />
        </svg>
      </div>
      testMenu2
    </a>
    <li>
      <p
        className="pt-3 pb-3 pl-5 text-sm text-gray-400 border-t border-gray-600 border-solid"
      >
        various test title
      </p>
    </li>
    <a
      className="text-white flex items-center pl-5 py-4 w-full text-left cursor-pointer font-medium text-lg hover:bg-zinc-600  text-white "
      data-testid=""
      href="/app/testMenu3"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="mr-4"
      >
        <svg
          fill="#FFFFFF"
          height={24}
          viewBox="0 0 512 512"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M457.6 1.5c-2.2.7-6.4 3.8-9.5 6.8l-5.6 5.5-3.9-2.9c-4.4-3.3-7.3-3.6-11.6-1.4-5.6 2.9-6.8 10.2-2.5 15.3 3 3.6 3.1 4 .4 6.9l-2.2 2.2-3.4-2.9c-4.8-4.2-10.4-4.2-14.4.1s-4 9.2.1 14l2.9 3.6-2.5 2.4c-4.1 3.8-6.4 9.2-6.4 15.1v5.3l-42.7 42.7-42.8 42.8-7.2-5.8c-15.7-12.4-32.2-21.2-47.9-25.3-9-2.3-24.8-2.8-32.9-1-18.6 4.2-35.6 16.4-43.7 31.2-2 3.6-5.5 11.7-7.8 18-12 31.9-16.6 35.7-52 42.5-27.5 5.2-42.4 9.7-56.9 17.2-11.5 5.9-21.9 13.2-31.2 22C-1.5 289-7.5 331.2 16.3 378.6c11.3 22.5 25.5 41.1 50 65.6 19.2 19.1 31.3 29.3 46.2 39.3 58.8 39 111.2 34.8 149.5-12 15.5-19 24.3-38.4 30.4-67.3 1.4-6.7 3.8-17.7 5.2-24.3 1.4-6.7 3.6-14.5 4.9-17.4 4.5-10.4 13.3-16.4 34.5-24 27.9-9.9 40.6-21.7 48.6-45 3.8-10.9 4-27.7.6-40.3-4.2-15.5-15.1-35.4-26.8-49.2l-4.6-5.3 42.9-42.9 42.8-42.8h5.3c5.9 0 11.3-2.3 15.1-6.4l2.4-2.5 3.6 2.9c4.9 4.1 10.1 4.1 14.2.1 4.1-4.1 4.1-9.7-.1-14.4l-2.9-3.4 2.2-2.2c2.9-2.7 3.3-2.6 6.9.4 5.1 4.3 12.4 3.1 15.3-2.5 2.2-4.3 1.9-7.2-1.4-11.6l-2.8-3.9 5.7-6.1c6.5-6.7 8.6-12.1 7.6-18.8-1-6.1-3.2-8.8-20.8-26.2C477.9 5.6 473.7 2 470.5 1.1c-5.2-1.4-8.1-1.3-12.9.4zm9.2 23.1c5 3.5 3.6 5.5-19 28.1-22.3 22.3-24.3 23.7-28.2 19.7-4-3.9-2.6-5.9 19.7-28.1 13.5-13.6 22-21.3 23.3-21.3 1.1 0 3 .7 4.2 1.6zm20.8 20.9c.9 1.4 1.4 3.4 1 4.7-.3 1.3-9.9 11.6-21.4 23-21.7 21.7-23.7 23.1-27.6 19.2-4-3.9-2.6-5.9 19.6-28.1 23-23 24.8-24.2 28.4-18.8zM256.7 154.2c9.9 3.2 21.2 9.5 31.5 17.5l6 4.6-32.9 32.9-32.8 32.8-6.8.5c-13.1.9-22.2 4.8-30.9 13.6-10.4 10.3-15.2 24.9-13.1 39 4.8 30.9 37.5 48.4 65.6 35.1 7.9-3.8 13.5-8.6 18.6-16.2 4.6-7 6.8-13.8 7.5-23.5l.5-6.9 32.9-32.9 32.9-32.9 7.3 9.9c13.1 17.7 18.5 31.8 17.8 46.8-.3 7.4-.9 10.1-3.4 15.5-5.6 11.9-11.6 16.5-29.4 22.8-21 7.4-30.6 12.8-39.9 22.7-6.3 6.8-11.6 16.6-14.5 26.7-.9 3.1-3.6 15.5-6.1 27.6-5.8 28.8-10.4 41.3-20.8 56.7-5.9 8.8-18.2 21.1-26.3 26.5-10.8 7.1-19.4 9.5-32.9 9.4-23.6-.3-47.3-10.6-75.5-33-11-8.7-40.7-38.4-49.4-49.4-22.4-28.2-32.7-51.9-33-75.5-.1-9.1.3-12.3 2.3-18.3 7.6-23.1 34.1-46.2 63.6-55.5 4.4-1.4 16.3-4.2 26.5-6.2 33.5-6.7 43.4-10.4 53.9-20.1 9.7-8.9 15.5-18.8 22.6-37.9 8.1-22.1 13.4-28.4 28.2-33.4 9-3.1 18.4-2.8 30 1.1z"
            fillRule="nonzero"
          />
          <path
            d="M98.8 328.8c-5.4 5.5-7.8 8.6-7.8 10.3 0 3.4 78.4 81.9 81.8 81.9 1.6 0 5-2.6 10.3-7.8 6.3-6.2 7.9-8.3 7.9-10.7 0-2.6-5.3-8.3-39.3-42.2-33.7-33.8-39.6-39.3-42.1-39.3-2.4 0-4.6 1.6-10.8 7.8z"
            fillRule="nonzero"
          />
        </svg>
      </div>
      testMenu3,testMenu4
    </a>
  </ul>
  <button
    className="absolute bottom-0 flex justify-end text-right pr-5  w-full text-white border-t border-gray-600 border-solid hover:opacity-70 py-7"
    data-testid="toggleBtn"
    onClick={[Function]}
  >
    <svg
      fill="white"
      height={20}
      viewBox="0 0 306 306"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M247.35 35.7 211.65 0l-153 153 153 153 35.7-35.7L130.05 153z"
        fillRule="nonzero"
      />
    </svg>
  </button>
</aside>
`;
